### PR TITLE
chore(ununpack): Remove external files

### LINKED ISSUE #1462: Remove additional C files from ununpack agent
### Description

OpenSSL provides functions to calculate different digests MD5, SHA1 and SHA256.

In this way, we can remove additional files and make it as a library dependency.

https://github.com/openssl/openssl/tree/OpenSSL_1_0_1/crypto/sha
https://github.com/openssl/openssl/tree/OpenSSL_1_0_1/crypto/md5

### Versions

* Last commit id on master: a0de8e9490a69b03e6d831e13d188e409f9845fd

### PR DESCRIPTION
**Pull request by GMishx**
## Description

Remove external files from ununpack agent which are used to calculate the checksum of files.

Also, remove OpenSSL dependency from delagent due to recent changes in OpenSSL license.

See #1462 for more info.

### Changes

1. Remove external C and H files from ununpack agent.
1. Use [libgcrypt](https://dev.gnupg.org/source/libgcrypt/) instead (LGPL-2.1).
1. Remove OpenSSL dependency from delagent, use `libgcrypt`.

## How to test

1. Check if ununpack and wgetagent works during file upload.
1. Check if delagent works (both from CLI and UI).

Closes #1462

### CODE REVIEW

**Comment by shaheemazmalmmd** [comment]
**Tested :**
* With large package for ununpack agent. works fine.
* Upload zlib package from URL to test the wget agent. works fine.
* Executed some delete options using delagent. works fine(but number of items processed shows zero as count which is confusing).

File: src/ununpack/agent/checksum.c [hunk 1]
```diff
@@ -126,22 +126,30 @@ int	CountDigits	(uint64_t Num)
...
-  rc = SHA1Reset(&sha1);
-  if (rc)
+  checksumError = gcry_md_open(&checksumhandler, GCRY_MD_NONE, 0);
+  if (! checksumhandler)
+  {
+    LOG_ERROR("Unable to initialize checksum\n");
+    free(Sum);
+    return(NULL);
+  }
+  checksumError = gcry_md_enable(checksumhandler, GCRY_MD_MD5);
```
Comment (ag4ums) [inline src/ununpack/agent/checksum.c]: check for the error code here as well 
 if (gcry_err_code(checksumError) != GPG_ERR_NO_ERROR)

File: src/ununpack/agent/checksum.c [hunk 2]
```diff
@@ -183,34 +185,39 @@ Cksum *	SumComputeFile	(FILE *Fin)
...
   {
-    LOG_ERROR("Failed to compute sha1\n");
-    free(Sum);
+    LOG_ERROR("GCRY Error: %s/%s\n", gcry_strsource(checksumError),
+        gcry_strerror(checksumError));
     return(NULL);
   }
+
+  Sum = (Cksum *)calloc(1,sizeof(Cksum));
+  if (!Sum) return(NULL);
```
Comment (ag4ums) [inline src/ununpack/agent/checksum.c]: if (! Sum) { ... }

File: src/ununpack/agent/checksum.c [hunk 3]
```diff
@@ -243,3 +250,39 @@ char *	SumToString	(Cksum *Sum)
...
+  unsigned char *tempBuff;
+  memset(buf, '\0', sizeof(buf));
+  FILE *f;
+  if(!(f=fopen(filename, "rb")))
+  {
+    LOG_FATAL("Failed to open file '%s'\n", filename);
+    return(1);
+  }
+  gcry_md_open(&checksumhandler, GCRY_MD_SHA256, 0);
+  if (! checksumhandler)
```
Comment (ag4ums) [inline src/ununpack/agent/checksum.c]: int gcry_md_is_enabled (gcry_md_hd_t h, int algo) seems to be a better check

File: src/ununpack/agent/checksum.c [hunk 4]
```diff
@@ -183,34 +185,39 @@ Cksum *	SumComputeFile	(FILE *Fin)
...
+    LOG_ERROR("Unable to initialize checksum\n");
     return(NULL);
   }
-
-  MyMD5_Update(&md5,CF->Mmap,CF->MmapSize);
-  MyMD5_Final(Sum->MD5digest,&md5);
-  SHA1Input(&sha1,CF->Mmap,CF->MmapSize);
-  rc = SHA1Result(&sha1,Sum->SHA1digest);
-  if (rc)
+  checksumError = gcry_md_enable(checksumhandler, GCRY_MD_MD5);
```
Comment (ag4ums) [inline src/ununpack/agent/checksum.c]: check for the error code here as well
if (gcry_err_code(checksumError) != GPG_ERR_NO_ERROR)

File: src/ununpack/agent/checksum.c (hunk 1 above)
Comment (GMishx) [reply src/ununpack/agent/checksum.c]: Done, please check.

File: src/ununpack/agent/checksum.c (hunk 2 above)
Comment (GMishx) [reply src/ununpack/agent/checksum.c]: Done, please check.

File: src/ununpack/agent/checksum.c (hunk 3 above)
Comment (GMishx) [reply src/ununpack/agent/checksum.c]: Done, please check.

File: src/ununpack/agent/checksum.c (hunk 4 above)
Comment (GMishx) [reply src/ununpack/agent/checksum.c]: Done, please check.

**Pull request merged**